src/backend/services/agent.py

```diff
@@ -42,3 +44,37 @@ def raise_db_error(e: Exception, type: str, name: str):
         )
 
     raise HTTPException(status_code=500, detail=str(e))
+
+
+def get_public_and_private_agents(
+    session: DBSessionDep,
+    user_id: str,
+    organization_id: Optional[str] = None,
+    offset: int = 0,
+    limit: int = 100,
+) -> list[Agent]:
+    public_agents = agent_crud.get_public_agents(
```

Comment:
Could we have one function with a filter for private?

Reply:
Do you mean a crud function?